docs/howto/krustlet-on-azure.md

```diff
@@ -0,0 +1,70 @@
+# Running Krustlet on Azure
+
+This guide demonstrates how to run Krustlet on Azure.
+
+## Prerequisites
+
+This guide will require both [the Azure
+CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) as well as
+`kubectl` to connect to the cluster after it has been provisioned.
+
+This specific tutorial will run Krustlet on another Azure Virtual Machine within
+the same Virtual Network as the Kubernetes cluster.
+
+## Step 1: Creating a Service Principal
+
+Prior to deploying Krustlet, a Service Principal needs to exist.
+
+The following Azure CLI command can be used to create a Service Principal:
+
+```shell
+$ az ad sp create-for-rbac --name ServicePrincipalName --skip-assignment
+```
+
+The output for a service principal with password authentication includes the
+password key. Make sure you copy this value - it can't be retrieved. If you
+forget the password, [reset the service principal
+credentials](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli#reset-credentials).
+
+The ID and password appears in the output of `az ad sp create-for-rbac` and are
+used in the ARM template's parameters. Make sure to record their values for
+later use.
+
+## Step 2: Generate an SSH Key
+
+You will also need to generate an SSH key. This will be used to SSH into the
+machines for debugging purposes... Or if you're just curious and want to see
+[how the sausage is
+made](https://en.wiktionary.org/wiki/how_the_sausage_gets_made).
+
+Follow the guide on Github to [generate a new SSH
+key](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
+
+## Step 3: Click the Button
+
+Go ahead. Click the button.
+
+[![Deploy To
+Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fbacongobbler%2Fkrustlet-on-azure%2Fmain%2Fazuredeploy.json)
```

Comment:
I think we should put this `azuredeploy.json` in this repo rather than have it live it a separate one. Or we should move your repo under the DeisLabs org

Reply:
Sounds good!

Reply:
1.19.3 is still in preview for AKS, so I'll use 1.18.10 instead.

Reply:
(according to `az aks get-versions -l westus2` as of this morning)